centinel/as_info.py

```diff
@@ -0,0 +1,64 @@
+#
+# Abbas Razaghpanah (arazaghpanah@cs.stonybrook.edu)
+# March 2015, Stony Brook University
+#
+# as_info.py: a library to lookup AS number and AS owner information
+# for IP addresses.
+#
+
+from netaddr import IPNetwork, IPAddress
```

Comment:
I use a BGP database on a [website](http://thyme.apnic.net/) hosted by APNIC that curates the table using routing information from its routers. What I do in this library, is matching the IP address to the longest prefix on the table and get the AS info from the other table.
Getting AS numbers from BGP tables is as accurate as it gets (though I'm not sure about AS number to AS name/info).